app/models/solidus_paypal_braintree/transaction_address.rb

```diff
@@ -51,16 +51,24 @@ def to_spree_address
         zipcode: zip
       )
 
-      if !first_name.nil?
-        ::Spree::Deprecation.warn("first_name and last_name are deprecated. Use name instead.", caller)
-        address.first_name = first_name
-        address.last_name = last_name || "(left blank)"
-      elsif address.respond_to? :name
-        address.name = name
+      if SolidusSupport.combined_first_and_last_name_in_address?
+        address.name = begin
+          if first_name.nil?
```

Comment:
Why this check? In Solidus 2.11 we'll have both `first_name` and `name` always synced and this is probably always false. In Solidus 3, there's no reason to check `first_name`, which is no more used.

Reply:
Here `first_name` is an instance attribute of `SolidusPaypalBraintree::TransactionAddress` (together with `last_name` and `name`). If I recall correctly these values are returned by Braintree itself and maybe they can be valorized differently according to the API version (that's why we need the check on its presence).

Reply:
Ah makes sense!